tests/aspnetcore/runner/Program.cs

```diff
@@ -0,0 +1,155 @@
+        {
+	    bool hasUnitTest = false;
+	    bool failed = false;
+	    Assembly assembly = System.Reflection.Assembly.LoadFrom(fullpath);
+	    foreach(Type type in assembly.GetTypes())
+	    {
+		if (type.IsAbstract)
+		    continue;
+
+		if (type.GetConstructor(Type.EmptyTypes) == null)
+		    continue;
+
+		//Console.WriteLine("Class {0}", type);
+		foreach (MethodInfo method in type.GetRuntimeMethods())
+		{
+		    bool isUnitTest = false;
+		    List<IList<CustomAttributeTypedArgument>> theoryArgs = new List<IList<CustomAttributeTypedArgument>>();
+		    //Console.WriteLine("    Method: {0}", method.Name);
+		    foreach(var attr in method.CustomAttributes)
+		    {
+			//Console.WriteLine("        custom attribute: {0} with {1} Constructor Arguments", attr.AttributeType.Name, attr.ConstructorArguments.Count);
+			if (attr.AttributeType.Name == "FactAttribute" || attr.AttributeType.Name == "TheoryAttribute")
+			{
+			    isUnitTest = true;
+			}
+
+			if (attr.AttributeType.Name=="InlineDataAttribute")
+			{
+			    theoryArgs.Add(attr.ConstructorArguments);
+			}
+		    }
+
+		    if (isUnitTest)
+		    {
+			hasUnitTest = true;
+			var obj = Activator.CreateInstance(type);
+			try
+			{
+			    if (theoryArgs == null)
+			    {
+				//Console.WriteLine("--- Running test " + method.Name + " with no parameters");
+				method.Invoke(obj, null);
+			    }
+			    else
+			    {
+				foreach (var theoryArg in theoryArgs)
+				{
+				    List<object> pprams = new List<object>();
+				    foreach (var onearg in theoryArg)
+				    {
+					if (onearg.Value != null && onearg.Value.GetType() == typeof(ReadOnlyCollection<CustomAttributeTypedArgument>))
+					{
+					    var argList = (ReadOnlyCollection<CustomAttributeTypedArgument>)onearg.Value;
+					    foreach (var myarg in argList)
+					    {
+						//Console.WriteLine("Add parameter in collection {0}", myarg.Value);
+						object argValue = getObjOrArrayValue(myarg);
+						pprams.Add(argValue);
+					    }
+					}
+					else {
+					    //Console.WriteLine("Add parameter {0}", onearg.Value);
+					    pprams.Add(onearg.Value);
+					}
+				    }
+				    string msg = "--- Running test " + method.Name + " with parameters (";
+				    foreach (var o in pprams)
+				    {
+					msg += (o==null ? "null" : o.ToString()) + " ";
+				    }
+				    //Console.WriteLine(msg + ")");
+
+				    method.Invoke(obj, pprams.ToArray());
+				}
+			    }
+			}
+			catch(Exception e)
+			{
+			    //Console.WriteLine("Failure");
+			    Console.WriteLine(e);
+			    failed = true;
+			}
+		    }
+		}
+	    }
+	    if (hasUnitTest)
+	    {
+		Console.WriteLine("===== {0}! {1}", failed ? "Failed" : "Passed", assemblyName);
+	    }
+        }
+
+        static void Main(string[] args)
+        {
+            if (args.Length != 2 && args.Length != 1)
+            {
+                Console.WriteLine("Usage: runner <file-with-list-of-tests> <path-prefix>");

```

Comment:
Whats the path-prefix option for?

Reply:
Xuejun added it. It shortens the test paths by eliminating the prefix. We can retain this for now.